doc/developers/contributing.rst

```diff
@@ -762,6 +762,86 @@ To test code coverage, you need to install the `coverage
 
 3. Loop.
 
+Monitoring performance [*]_
+===========================
+
+When proposing changes to the existing code base, it's important to make sure
+that they don't introduce performance regressions. Scikit-learn uses
+`asv benchmarks <https://github.com/airspeed-velocity/asv>`_ to monitor the
+performance of a selection of common estimators and functions. The benchmark
+suite can be found in the `scikit-learn/asv_benchmarks` directory.
+
+To use all features of asv, you will need either `conda` or `virtualenv`. For
+more details please check the `asv installation webpage
+<https://asv.readthedocs.io/en/latest/installing.html>`_.
+
+First of all you need to install the development version of asv::
+
+  pip install git+https://github.com/airspeed-velocity/asv
+
+and change your directory to `asv_benchmarks/`. The benchmark suite is
+configured to run against your local clone of scikit-learn. Make sure it is up
+to date::
+
+  git fetch upstream
+
+In the benchmark suite, the benchmarks are organized following the same
+structure as scikit-learn. For example, you can run the benchmarks for the
+`linear_model` module only::
+
+  asv run -b linear_model upstream/master^!
+
+or for a specific estimator::
+
+  asv run -b LogisticRegression upstream/master^!
+
+You can run the benchmarks against any commit, the branch you are working on
+for example::
+
+  asv run -b linear_model my_branch^!
+
+Simply remove the `-b` flag to run the full benchmark suite::
+
+  asv run my_branch^!
+
+However this can take up to two hours. The `-b` flag also accepts a regular
+expression for a more complex subset of benchmarks to run.
+
+You can compare the performances of two commits. For example, comparing master
+and the branch you are working on::
+
+  asv continuous upstream/master my_branch
```

Comment:
Actually I think we should move the comparison case as the first example but on a specific estimator:

```
asv continuous -b LogisticRegression upstream/master HEAD
```

and then present the other commands.